controllers/managedosversionchannel_controller.go

```diff
@@ -204,18 +204,22 @@ func (r *ManagedOSVersionChannelReconciler) reconcile(ctx context.Context, manag
 		return ctrl.Result{}, err
 	}
 
-	return r.handleSyncPod(ctx, pod, managedOSVersionChannel, interval)
+	return r.handleSyncPod(ctx, pod, managedOSVersionChannel, interval), nil
```

Comment:
Once the pod started we never return error, at most track the failed status in ready condition. This is to prevent error infinite loops. If we return an error here we directly trigger a new reconcile loop which will create a new attempt to sync.